src/Yoti.Auth/Verifications/AgeVerificationParser.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Yoti.Auth.Verifications
+{
+    internal class AgeVerificationParser
+    {
+        private Dictionary<string, AgeVerification> _ageUnderVerificationsDict;
+        private Dictionary<string, AgeVerification> _ageOverVerificationsDict;
+        private List<AgeVerification> _allVerificationsDict;
+
+        public AgeVerificationParser(BaseProfile baseProfile)
+        {
+            _ageUnderVerificationsDict = FindAgeUnderVerifications(baseProfile);
+            _ageOverVerificationsDict = FindAgeOverVerifications(baseProfile);
+
+            _allVerificationsDict = _ageUnderVerificationsDict.Values
+                .Concat(_ageOverVerificationsDict.Values).ToList();
+        }
+
+        public ReadOnlyCollection<AgeVerification> FindAllAgeVerifications()
+        {
+            return _allVerificationsDict.AsReadOnly();
```

Comment:
We can make this _AsReadOnly_ call in the constructor - so then we don't have to create a new instance each time it's called